### DISCUSSION TITLE
Duty Cycle for Dew Heaters

### DISCUSSION THREAD

**Discussion by fly800** [aaronwmorris/indi-allsky#1524] [Q&A]
Can anyone tell me, what is the duty  cycle setting in Device Config used for?  I want to control a resistor bank dew heater using a mosfet.  The mosfet is controlled by using pin D12 PWM.  I tested the output of D12 with an oscilloscope and get 3.3vdc with setting at 100% duty cycle.  Isn't the output variable depending on the climate conditions.  If so, should I see PMW pulses. Is there a way to test this?

**Comment by aaronwmorris** [comment]
~~You are not the first to have issues with the PWM.  I am currently testing to find out why it is not working.~~

**Comment by aaronwmorris** [comment]
I hooked up a scope to my Pi5 and I see proper PWM signals.  If you set a duty cycle of 100%, the pin will effectively be set to HIGH.  In order to see the PWM, the setting would have to be somewhere between 1 and 99.

**Comment by fly800** [comment]
I got the dew heater up and running using PMW.  Winter is coming and I need to address external dome frost. What do you recommend as sensors' locations to better control dew?  I have a BME680 sensor (temp, RH and Pressure) in the housing.  I can use OpenWeather to get exterior temp and humidity.  I have the housing open to outdoors but the difference in temperatures is approx 12deg C and humidity 30%.  Numbers are higher in the housing.  This is because of the heat generated by the RPi and also for the housing to become equilibrium with outside.  

Thanks for your help

**Reply by aaronwmorris** [reply]
The temperature and humidity definitely need to be read from outside.  This is just not a problem I have given much thought to.  There probably needs to some sort of antechamber that allows the sensors to be exposed to the outside air without exposing the inside of the case to the outside air.

**Reply by fly800** [reply]
Thank you for your suggestions. I have been dealing with condensation inside the dome and have tried various methods to seal it, but there is always a leak. One option is to use desiccants, but they would need to be replaced regularly. Another approach is to maintain the same temperature inside as outside. To achieve this, I have a partition with a fan separating the sensors and the RPi, as well as a 1/2" hole at the bottom of the enclosure. I might need to place a fan on the intake, but I am concerned about water being drawn in.

**Reply by bluthen** [reply]
I've had success with rubber gaskets between every joining of materials. 

I have also went with having a simple temperature sensor and control board outside, 10K NTC and opamp + power mosfet with selected resistor values that define when to turn on the heater, so I don't need to worry about running a control cable from the Pi to the heater. The control board can be conformal coated just in case.

I was mainly concerned about running the heater when it is already hot. So I selected values to have it turn the heater on when below 70F and turn off when above 80F, a 10F hysteresis. Ideal would be to include humidity, however humidity sensors usually can be damaged by too much moisture, by their nature can not be sealed from the air, and maybe need more electronics to support them.

**Reply by aaronwmorris** [reply]
I would strongly encourage liberal application of silicone (or caulk).  Make your own gaskets, effectively.

I will apply silicone around every junction of the enclosure, even the already supposedly water proof cable glands.  One of my enclosures kept having internal dewing after it rained, and I finally filled in the top recessed screw holes, even though the recessed holes were outside of the internal gasket of the enclosure.  Water is pervasive.

Just do not make the mistake I made with my first enclosure.  I accidentally used silicone _sealant_ as a gasket for the dome, which hardened like glue.  I had to replace the entire enclosure because I could never remove that dome (it developed cracks).  **Use regular silicone**.

**Reply by bendenajones** [reply]
I live in a heavy dew location (near CA coast) and I use a small 5v 40mm2 exhaust fan; A small inlet (25mm diameter) hole with a metal screen to keep the little critters out on the other side of the dome bottom has worked well for me. I use those Polyimide Heaters for my Dew heater; two stripes opposite on the inside of the dome protected from the sun by thin copper stripes. I've had no internal moisture issues whatsoever and no dew. 
![image](https://github.com/user-attachments/assets/8db2f5e1-4969-4281-8b7f-52cc4e9fd6ec)

![image](https://github.com/user-attachments/assets/e8807e47-67c8-4862-ba2a-d86cd21a602d)
